### ISSUE TITLE
Learning rate Scheduler

### ISSUE THREAD

**Issue by luistelmocosta** [qubvel/segmentation_models.pytorch#115]
Is there any example of a lr_scheculer using SMP? 

regards

**Comment by jchen42703** [comment]
SMP provides models and loss functions. LR Schedulers should be separate from SMP. Could you elaborate on what you mean?

**Comment by julienguegan** [comment]
> SMP provides models and loss functions. LR Schedulers should be separate from SMP. Could you elaborate on what you mean?

I am also interested in this issue. For example, in PyTorch, when you want to configure a learning rate scheduler you add `lr_scheduler.step()` after you `optimizer.step()` in your **training loop** inside the **dataloader loop** but here with SMP, you only call `train_epoch.run(train_loader)` and there is no way to add a scheduler in argument of this function.

Do you have a way of doing it ?

**Comment by seanbenhur** [comment]
Any of you found any workaround for this?

**Comment by marekpaulik** [comment]
Same question as above. Any of you found any workaround for this?

**Comment by nbaker47** [comment]
Same question... Anyone solved?

**Comment by ckolluru** [comment]
If we want a learning rate scheduler during the train epochs, do we just edit lines 73 to 97 in utils\train.py?

Portions of code below that are in between ** signs are the additions that I think might be sufficient.

Can someone please check?

```
class TrainEpoch(Epoch):
    def __init__(self, model, loss, metrics, optimizer, **scheduler=None**, device="cpu", verbose=True):
        super().__init__(
            model=model,
            loss=loss,
            metrics=metrics,
            stage_name="train",
            device=device,
            verbose=verbose,
        )
        self.optimizer = optimizer
        **self.scheduler = scheduler**

    def on_epoch_start(self):
        self.model.train()

    def batch_update(self, x, y):
        self.optimizer.zero_grad()
        prediction = self.model.forward(x)
        loss = self.loss(prediction, y)
        loss.backward()
        self.optimizer.step()
        **if self.scheduler is not None:
            self.scheduler.step()**
        return loss, prediction
```
